### DISCUSSION TITLE
PyCharm plugin for quarto?

### DISCUSSION THREAD

**Discussion by baggiponte** [quarto-dev/quarto-cli#1680] [Feature Requests]
Is there a PyCharm plug-in (much alike VS code’s) on your roadmap? I’d love to help - unfortunately I have never developed PyCharm/JetBrains plug-ins, but I can open and follow an issue on their YouTrack page.

**Comment by jjallaire** [comment]
We have the creation of a proper Quarto LSP server on our roadmap (which would extract a bunch of the VS Code work out into a standalone binary). This could then be hooked up to PyCharm via <https://plugins.jetbrains.com/plugin/10209-lsp-support>

**Reply by matuladominik** [reply]
It has been a while, is there any progress 🙏 ?

**Reply by Iakl** [reply]
Im also looking forward for it

**Reply by ntluong95** [reply]
I am looking for this feature also @jjallaire

**Reply by danieltomasz** [reply]
Even rudimentary plugin (like register qmd as markdown and running  cli commands  (render/preview) via pycharm command paletter)will be much of help, I dont expected the level of support that VSCode get :)

**Reply by danieltomasz** [reply]
@jjallaire the plugin you linked is outdated https://blog.jetbrains.com/platform/2023/07/lsp-for-plugin-developers/ but

>Starting with the 2023.2 release cycle, the LSP API is publicly available as part of the IntelliJ Platform

**Reply by mcanouil** [reply]
@danieltomasz The plugin linked was linked a year ago. 😉

**Comment by ethantenison** [comment]
Really hoping for this 🥺

**Reply by ntluong95** [reply]
You can upvote here as well: https://youtrack.jetbrains.com/issue/R-1440/Quarto-support-successor-to-Markdown

**Comment by CodrinSocol** [comment]
Any updates on this?

**Reply by mcanouil** [reply]
We appreciate your interest in the project, but I'd like to share some guidance on how we can make discussions more productive.

Posts asking "Any news?" or "When will this be ready?" without specific questions or constructive feedback don't help move development forward.
They create notification noise for maintainers without providing actionable input that could influence development priorities.

Rather than asking for general updates, consider asking specific technical questions about implementation, contributing implementation suggestions, or submitting pull requests for features you need.
Consider upvoting and/or adding a "thumb up" on the original message.

The most valuable contributions focus on "here's how I can help solve this" rather than "here's what I want you to do for me."

For the latest information about progress and planned features, check the project's changelog, release notes, or issues.

Thanks for understanding and for your continued interest in the project.
